fortls/helper_functions.py

```diff
@@ -68,8 +71,15 @@ def detect_fixed_format(file_lines: list[str]) -> bool:
     Lines wih ampersands are not fixed format
     >>> detect_fixed_format(['trailing line & ! comment'])
     False
+
+    But preprocessor lines might be ignored
+    >>> detect_fixed_format(['#if defined(A) && !defined(B)'], preproc=True)
+    True
     """
     for line in file_lines:
+        # Ignore preprocessor lines
+        if preproc and line.startswith("#"):
+            continue
```

Comment:
This should be present at all times. I would suggest to remove the optional argument `preproc` and have the statement present always.
What about multi-line preprocessor directives?

Reply:
I will remove the `preproc` flag. I added it because I wasn't sure whether there can be valid Fortran code that starts with a `#` (especially since I imagine to have heard that some compilers accept all kinds of characters in the first column of fixed-form Fortran to indicate comments, but wasn't sure whether `#` is one of them), so I just thought I go with the more conservative implementation.

About the multi-line directives: Do you think it is sufficient to consider lines ending in `\` for multi-line directives? I think I have read somewhere that some Fortran compilers support other ways of indicating multi-line characters, but wasn't sure how `fortls` wants to handle this (if I read the code correctly, I would say that `fortls` currently only handles `\` when post-processing files as well).